### ISSUE TITLE
productionon build failed for ionic3 and angular4

### ISSUE THREAD

**Issue by mburger81** [driftyco/ionic-app-scripts#877]
#### Short description of the problem:
We migrated today our app to ionic3 and angular4 but run in several issues for build production bundle.

Using ionic-app-script 1.3.0
`ionic cordova:build android --prod --release`
or
`npm run build --prod`

We get this error:
```
[10:40:07]  cleancss finished in 1.25 s 
[10:40:08]  ionic-app-script task: "build" 
[10:40:08]  Error: [object Object] 
Error: [object Object]
    at BuildError.Error (native)
    at new BuildError (/home/abc/software/workspace/web-client-ionic/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
    at ChildProcess.<anonymous> (/home/abc/software/workspace/web-client-ionic/node_modules/@ionic/app-scripts/dist/worker-client.js:38:24)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:191:7)
    at process.nextTick (internal/child_process.js:753:12)
    at _combinedTickCallback (internal/process/next_tick.js:73:7)
    at process._tickCallback (internal/process/next_tick.js:104:9) hasBeenLogged: false, isFatal: false, name: 'Error' }
```

Using latest nightly `@ionic/app-scripts": "1.3.0-201704051913` and run
`ionic cordova:build android --prod --release`
or
`npm run build --prod`

We get this error:
```
[10:31:58]  cleancss finished in 1.15 s 
{ Error: Unexpected token: name (DEFAULT_LOG_LEVEL) in /home/abc/software/workspace/web-client-ionic/www/build/main.js at line 156519, col 6, pos 6619073
    at BuildError.Error (native)
    at new BuildError (/home/abc/software/workspace/web-client-ionic/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
    at taskReject (/home/abc/software/workspace/web-client-ionic/node_modules/@ionic/app-scripts/dist/worker-process.js:36:22)
    at /home/abc/software/workspace/web-client-ionic/node_modules/@ionic/app-scripts/dist/worker-process.js:14:13
    at process._tickCallback (internal/process/next_tick.js:109:7) hasBeenLogged: false, isFatal: false, name: 'Error' }

```

#### What behavior are you expecting?
It should work :smile: 

I know I shouldn't do this, but it is urgent and so I would mention some one of you guys @danbucholtz

**Comment by danbucholtz** [comment]
@mburger81,

Are you bundling ES2015 code? Uglify seemingly cannot handle it. We were previously transpiling the bundle down to ES5 but that was proving to be slow. Maybe we need to turn that back on again.

Thanks,
Dan

**Comment by mburger81** [comment]
Probably yes, we are using a set of libraries which probably includes ES2015 code! We use this libs:
```
"@swimlane/ngx-datatable": "7.1.1",
"angular2-websocket": "0.9.1",
"chart.js": "2.5.0",
"jssip": "3.0.7",
"ng2-charts": "1.5.0",
```
and also includes this
`https://github.com/Streamedian/html5_rtsp_player`

what do you think?

**Comment by mburger81** [comment]
Some guys are discussing the issue on ionic issues
https://github.com/driftyco/ionic/issues/11092#issuecomment-292289896

**Comment by danbucholtz** [comment]
This should be resolved in `1.3.1`.

```
npm install @ionic/app-scripts@1.3.1
```

Thanks,
Dan

**Comment by mburger81** [comment]
I confirm, version 1.3.1 is "resolving" me the problem. (I think you added only a workaround)

**Comment by drewhawken** [comment]
@danbucholtz in 1.3.1 I am getting errors with minified and just plain js files I include in the build:
If I move appSettings.js up it will error out with appSettings.js.map not found, it's just plain javascript.

```
copyPolyfills: {
    src: ['{{ROOT}}/node_modules/ionic-angular/polyfills/polyfills.js',
      '{{ROOT}}/node_modules/atatus-js/atatus.min.js',
      '{{SRC}}/appSettings.js',
      '{{SRC}}/local.env.client.js',
      '{{ROOT}}/node_modules/pubnub/modern/dist/pubnub.min.js',
      '{{ROOT}}/node_modules/aws-sdk/dist/aws-sdk.min.js',
      '{{SRC}}/web-animations-next.min.js',
      '{{SRC}}/web-animations-next.min.js.map'
    ],
    dest: '{{BUILD}}'
  },

```

[10:28:25]  ionic-app-script task: "build"
[10:28:25]  Error: ENOENT: no such file or directory, open '/Users/drew/projects/mazu/www/build/atatus.min.js.map' in
            undefined at line undefined, col undefined, pos undefined
Error: ENOENT: no such file or directory, open '/Users/drew/projects/mazu/www/build/atatus.min.js.map' in undefined at line undefined, col undefined, pos undefined
    at BuildError.Error (native)
    at new BuildError (/Users/drew/projects/mazu/node_modules/@ionic/app-scripts/dist/util/errors.js:16:28)
    at taskReject (/Users/drew/projects/mazu/node_modules/@ionic/app-scripts/dist/worker-process.js:36:22)
    at /Users/drew/projects/mazu/node_modules/@ionic/app-scripts/dist/worker-process.js:14:13
    at process._tickCallback (internal/process/next_tick.js:103:7)

**Comment by danbucholtz** [comment]
@drewhawken, fixed in `1.3.2`.

Thanks,
Dan

**Comment by samastur** [comment]
Workaround version 1.3.1 works fine, 1.3.2 fails with unexpected token error (like described in #898).
